snooty/parser.py

```diff
@@ -287,9 +307,7 @@ def dispatch_visit(self, node: docutils.nodes.Node) -> None:
             ), f"Too many ids in this node: {self.docpath} {node}"
             if not node["ids"]:
                 self.diagnostics.append(
-                    Diagnostic.error(
-                        "Links must provide a valid URL", util.get_line(node)
-                    )
+                    InvalidURL("Links must provide a valid URL", util.get_line(node))
```

Comment:
Any reason why this message is not attached to the class?

Reply:
I originally only attached message construction logic to the classes where the message was invariable and there was more than one instance of that error (because I was mostly concerned with DRY principle). But after reading your comment and checking in with Andrew, I will now go ahead and attach the message to classes where the message is invariable :)